velox/exec/tests/MergingVectorOutputTest.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/MergingVectorOutput.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::exec::test {
+
+class MergingVectorOutputTest : public testing::Test,
+                                public velox::test::VectorTestBase {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+  void SetUp() override {
+    mergingVectorOutput_ =
+        std::make_shared<MergingVectorOutput>(pool(), 10UL << 20, 1024, 16);
+  }
+
+  std::shared_ptr<MergingVectorOutput> mergingVectorOutput_;
+};
+
+TEST_F(MergingVectorOutputTest, bufferSmallVector) {
+  vector_size_t batchSize = 10;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 110; ++i) {
```

Comment:
Maybe you shoud use a random number here so that we check with different batch sizes.

Reply:
Done.